src/routes/SafeAppLandingPage/SafeAppLandingPage.tsx

```diff
@@ -0,0 +1,321 @@
+        try {
+          const safeAppDetailsFromManifest = await getAppInfoFromUrl(safeAppUrl)
+          setIsManifestLoading(false)
+          setSafeAppDetailsFromManifest(safeAppDetailsFromManifest)
+        } catch (e) {
+          setIsManifestLoading(false)
+          setIsManifestError(true)
+        }
+      }
+
+      fetchSafeAppDetailsFromManifest()
+    }
+  }, [isManifestLoading, safeAppDetailsFromManifest, safeAppUrl, isManifestError])
+
+  const safeAppDetails = safeAppDetailsFromConfigService || safeAppDetailsFromManifest
+  const isLoading = isConfigServiceLoading || isManifestLoading
+
+  // redirect to the Welcome page if the Safe App details are invalid
+  useEffect(() => {
+    const isSafeAppMissing = !isLoading && appList.length > 0 && !safeAppDetails
+
+    if (isSafeAppMissing && isManifestError) {
+      history.push(WELCOME_ROUTE)
+    }
+  }, [isLoading, appList, safeAppDetails, isManifestError])
+
+  const availableChains = safeAppDetails?.chainIds
+  const showAvailableChains = availableChains && availableChains.length > 0
+
+  const userSafe = getUserSafe()
+
+  const showLoader = isLoading || !safeAppDetails
+
+  return (
+    <Container>
+      <StyledCard>
+        {showLoader ? (
+          <LoaderContainer>
+            <Loader size="md" />
+          </LoaderContainer>
+        ) : (
+          <>
+            {/* Safe App details */}
+            {safeAppDetails && (
+              <DetailsContainer>
+                <SafeIcon src={safeAppDetails.iconUrl} />
+                <DescriptionContainer>
+                  <SafeAppTitle size={'sm'}>{safeAppDetails.name}</SafeAppTitle>
+                  <div>{safeAppDetails?.description}</div>
+                </DescriptionContainer>
+              </DetailsContainer>
+            )}
+            <Separator />
+
+            {/* Available chains */}
+            {showAvailableChains && (
+              <>
+                <ChainLabel size="lg">Available networks</ChainLabel>
+                <ChainsContainer>
+                  {availableChains.map((chainId) => (
+                    <div key={chainId}>
+                      <NetworkLabel networkInfo={getChainById(chainId)} />
+                    </div>
+                  ))}
+                </ChainsContainer>
+                <Separator />
+              </>
+            )}
+
+            <ActionsContainer>
+              <UserSafeContainer>
+                <Title size={'xs'}>Use the dapp with your Safe!</Title>
+                {isWalletConnected ? (
+                  <>
+                    {userSafe ? (
+                      <div>not implemented</div>
+                    ) : (
+                      <>
+                        {/* Create new Safe */}
+                        <BodyImage>
+                          <Img alt="Vault" height={92} src={SuccessSvg} />
+                        </BodyImage>
+
+                        <Button
+                          data-testid={'create-new-safe-link'}
```

Comment:
You don't need braces around a string.

```suggestion
                          data-testid='create-new-safe-link'
```

Can you avoid using test IDs and select by the text shown?

Reply:
done